### ISSUE TITLE
Non-breaking spaces added inside HTML tags

### ISSUE THREAD

**Issue by jonathanpatt** [matthewlein/jQuery-widowFix#2]
I've been noticing a number of nbsps being added within the opening tags of HTML elements, for example:

```
<img src="..." &nbsp;alt="..." />
```

In most browsers, this doesn't break anything, but in IE it tends to break that element and cause (in the case of the img tag) a broken image to show instead. I've also noticed this occurring in form elements—for example, causing a hidden field to be visible.

**Comment by joshuaiz** [comment]
This is kind of a big issue - getting this behavior in a lot of unwanted places. I think a good solution would be to have widowFix ignore elements within html tags.

If you have widowFix set to run on say, `.entry-content p` then anything within html tags within <p> tags will get extra &nbsp; added. There are lots of times <p> elements will have html within: links, text formatting, etc. This is super no bueno.

If you have to start getting super specific with your selectors then it kind of defeats the purpose of this plugin.

**Comment by joakimc** [comment]
Breaks inline css.
`<div class="indicator" style="background-color: #9d612f"></div>`
is turned into
`<div class="indicator" style="background-color:&nbsp;#9d612f"></div>`
which won't render anywhere afaik.

**Comment by matthewlein** [comment]
I agree that this is a not perfect solution, and I'm open to PRs with tests that show its fixed :smile:

**Comment by tomasdev** [comment]
I recommend not using this plugin. Here's an alternative simpler method that doesn't break your HTML:

```javascript
function nbsp(html) {
    var text = $(html).text().replace(/\n/g, '').split(' ').filter(Boolean),
        secondLastWord = text[text.length - 2],
        lastWord = text[text.length - 1];

    return (
        html.substring(html.lastIndexOf(secondLastWord), 0, secondLastWord.length) +
        secondLastWord + '&nbsp;' +
        html.substring(html.lastIndexOf(secondLastWord) + secondLastWord.length)
    ).replace(secondLastWord + '&nbsp; ', secondLastWord + '&nbsp;').replace(' ' + lastWord, lastWord);
}
```
Sample covered cases:
```javascript
nbsp(`<h3 class="h1 divider">
        Get a closer <span class="nowrap">close-up.</span>
      </h3>`);
nbsp(`<h3 class="h1 divider">
        Get a closer close-up.<img src="" alt="some text" />
      </h3>`);
nbsp(`<h3>Get a closer closeup.<div style="background: #fff;"></div></h3>`);
nbsp(`<h3>Get a closer close-up.</h3>`);
nbsp(`<h3>Get a closer closeup.</h3>`);
```
